draft-ietf-masque-h3-datagram.md

```diff
@@ -181,74 +181,82 @@ all cases, the maximum permitted value of the H3_DATAGRAM SETTINGS parameter is
 
 ~~~
-  Datagram-Flow-Id = sf-list
+REGISTER_DATAGRAM_FLOW_ID Frame {
+  Type (i) = TBD,
+  Length (i),
+  Flow ID (i),
+  Encoded Field Section (..),
+}
 ~~~
+{: #register-frame-format title="REGISTER_DATAGRAM_FLOW_ID HTTP/3 Frame Format"}
 
-The "Datagram-Flow-Id" header field is used to associate one or more datagram
-flows with an HTTP message. As a simple example using a single
-flow, the definition of an HTTP method could instruct the client to use
-its flow ID allocation service to allocate a new flow ID, and
-then the client will add the "Datagram-Flow-Id" header field to its request to
-communicate that value to the server. In this example, the resulting header
-field could look like:
+The Type and Length fields follows the definition of HTTP/3 frames from {{H3}}.
+The payload consists of:
 
-~~~
-  Datagram-Flow-Id = 2
-~~~
+Flow ID:
 
-List members are flow ID elements, which can be named or unnamed.
-One element in the list is allowed to be unnamed, but all but one elements
-MUST carry a name. The name of an element is encoded in the key of the first
-parameter of that element (parameters are defined in Section 3.1.2 of
-{{STRUCT-FIELD}}). Each name MUST NOT appear more than once in the list. The
-value of the first parameter of each named element (whose corresponding key
-conveys the element name) MUST be of type Boolean and equal to true. The value
-of the first parameter of the unnamed element MUST NOT be of type Boolean. The
-ordering of the list does not carry any semantics. For example, an HTTP method
-that wishes to use four datagram flows for the lifetime of its
-request stream could look like this:
+: The flow ID to associate with the QUIC stream that carried this frame.
 
-~~~
-  Datagram-Flow-Id = 42, 44; ecn-ect0, 46; ecn-ect1, 48; ecn-ce
-~~~
+Encoded Field Section:
+
+: QPACK-encoded fields. This allows extensions to optionally attach headers to
+their flow IDs.
+
+Note that these registrations are unilateral and unidirectional: the sender of
+the frame unilateraly defines the semantics it will apply to the datagrams it
+receives. If a mechanism using this feature wants to send datagrams of a given
+flow ID in both directions, this frame will need to be exchanged in both
+directions.
 
-In this example, 42 is the unnamed flow, 44 represents the name
-"ecn-ect0", 46 represents "ecn-ect1", and 48 represents "ecn-ce". Note that,
-since the list ordering does not carry semantics, this example can be
-equivalently encoded as:
+
+# RELIABLE_DATAGRAM HTTP/3 Frame Definition {#reliable-datagram-frame}
+
+The RELIABLE_DATAGRAM frame (type=TBD) is used to send datagrams over QUIC
+streams when QUIC datagrams are unavailable or undesirable. Datagrams
+transmitted over streams using this frame have the same semantics as datgrams
+sent over the QUIC DATAGRAM frame.
 
 ~~~
-  Datagram-Flow-Id = 44; ecn-ect0, 42, 48; ecn-ce, 46; ecn-ect1
+RELIABLE_DATAGRAM Frame {
```

Comment:
What kinds of streams is this frame valid on?  If a flow ID can outlive a stream, how do you send it?